### PR TITLE
Add Parameter to transient transformation for Persistent GPU arrays

### PR DESCRIPTION
**Pull request by orausch**

### CODE REVIEW

File: daceml/onnx/onnx_importer.py [hunk 1]
```diff
@@ -371,7 +383,9 @@ def __call__(
         if self.do_auto_optimize:
             self.auto_optimize()
 
-        self.sdfg(**inputs, **outputs, **params, **symbols)
+        compiled = self.compile_and_init()
```
Comment (tbennun) [inline daceml/onnx/onnx_importer.py:386]: unrelated question: how do we cache after the SDFG is compiled, if at all? (i.e., between consecutive iterations)

File: daceml/pytorch/dlpack.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
+"""
+Resources:
```
Comment (tbennun) [inline daceml/pytorch/dlpack.py:2]: Maybe "Adapted from:"

File: tests/transformation/test_param_to_transient.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    dace_module.load_state_dict(pt_module.state_dict())
+
+    input = torch.rand(2, 10)
+    assert torch.allclose(pt_module(input), dace_module(input))
+
+    dace_module = DaceModule(dace_module, cuda=True)
+
+    assert torch.allclose(dace_module(input), pt_module(input))
+    parameter_to_transient(dace_module, "fc1.weight")
+    assert torch.allclose(dace_module(input), pt_module(input))
```
Comment (tbennun) [inline tests/transformation/test_param_to_transient.py:30]: what does this test do exactly? just ensure correctness is retained?

**Review by tbennun — APPROVED** [review]

File: tests/transformation/test_param_to_transient.py (hunk 3 above)
Comment (orausch) [reply tests/transformation/test_param_to_transient.py:30]: Yes

File: daceml/onnx/onnx_importer.py (hunk 1 above)
Comment (orausch) [reply daceml/onnx/onnx_importer.py:386]: We don't cache at the moment in daceml

**Pull request merged**